integrationtests/cli/apply/apply_online_test.go

```diff
@@ -0,0 +1,105 @@
+
+package apply
+
+import (
+	"github.com/golang/mock/gomock"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/rancher/fleet/integrationtests/cli"
+	"github.com/rancher/fleet/integrationtests/mocks"
+	"github.com/rancher/fleet/internal/client"
+	"github.com/rancher/fleet/internal/cmd/cli/apply"
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+
+	"github.com/rancher/wrangler/v2/pkg/generic/fake"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Fleet apply online", Label("online"), func() {
+
+	var (
+		ctrl             *gomock.Controller
+		getter           *mocks.MockGetter
+		c                client.Client
+		fleetMock        *mocks.FleetInterface
+		bundleController *fake.MockControllerInterface[*fleet.Bundle, *fleet.BundleList]
+		name             string
+		dirs             []string
+		options          apply.Options
+		oldBundle        *fleet.Bundle
+		newBundle        *fleet.Bundle
+	)
+
+	JustBeforeEach(func() {
+		//Setting up all the needed mocked interfaces for the test
+		ctrl = gomock.NewController(GinkgoT())
+		getter = mocks.NewMockGetter(ctrl)
+		c = client.Client{
+			Fleet:     mocks.NewFleetInterface(ctrl),
+			Namespace: "foo",
+		}
+		bundleController = fake.NewMockControllerInterface[*fleet.Bundle, *fleet.BundleList](ctrl)
+		fleetMock = c.Fleet.(*mocks.FleetInterface)
+		getter.EXPECT().GetNamespace().Return("foo").AnyTimes()
+		getter.EXPECT().Get().Return(&c, nil).AnyTimes()
+		fleetMock.EXPECT().Bundle().Return(bundleController).AnyTimes()
+		bundleController.EXPECT().Get("foo", gomock.Any(), gomock.Any()).Return(oldBundle, nil).AnyTimes()
+		bundleController.EXPECT().List(gomock.Any(), gomock.Any()).Return(&fleet.BundleList{}, nil).AnyTimes()
+	})
+
+	When("We want to delete a label in the bundle from the cluster", func() {
+		BeforeEach(func() {
+			name = "labels_update"
+			dirs = []string{cli.AssetsPath + "labels_update"}
+			//bundle in the cluster
+			oldBundle = &fleet.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"new":         "fleet-label2",
+						"new_changed": "fleet_label_changed",
+					},
+					Namespace: "foo",
+					Name:      "test_labels",
+				},
+			}
+			//bundle in the fleet.yaml file some values are autofulfilled in the implementation (this is why there are not only the labels)
+			newBundle = &fleet.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"new": "fleet-label2",
+					},
+					Namespace: "foo",
+					Name:      "test_labels",
+				},
+				Spec: fleet.BundleSpec{
+					Resources: []fleet.BundleResource{
+						{
+							Name:    "fleet.yaml",
+							Content: "labels:\n  new: fleet-label2",
+						},
+					},
+					Targets: []fleet.BundleTarget{
+						{
+							Name:         "default",
+							ClusterGroup: "default",
+						},
+					},
+				},
+			}
+		})
+
+		It("should correctly remove the labels from fleet.yaml", func() {
```

Comment:
Is this really what we validate?

Reply:
replace with "should delete the labels of the bunde present in the cluster that are no longer present in the fleet.yaml manifest" ?

Reply:
all right, I've changed it